code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -49,7 +49,8 @@
 
 /datum/reagent/proc/on_mob_life(mob/living/M)
 	current_cycle++
-	holder.remove_reagent(src.id, metabolization_rate * M.metabolism_efficiency) //By default it slowly disappears.
+	if(holder)
```

Comment:
The only time a reagent should be referrenced is by it's holder. Once that holder is nulled (which should only happen when it's GCd) the byond processor should automatically cache the reagent eventually. The alternative is deleting it outright, or adding in lines to shut down reagent processing when a holder is deleted.